src/OptionSettlementEngine.sol

```diff
@@ -57,90 +75,112 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
     mapping(uint160 => mapping(uint16 => bool)) internal _doesBucketIndexHaveUnexercisedOptions;
 
     /// @notice Accessor for mapping a claim id to its ClaimIndices
-    mapping(uint256 => ClaimIndex[]) internal _claimIdToClaimIndexArray;
+    mapping(uint256 => OptionLotClaimIndex[]) internal _claimIdToClaimIndexArray;
+
+    /*//////////////////////////////////////////////////////////////
+    //  Accessors
+    //////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc IOptionSettlementEngine
     function option(uint256 tokenId) external view returns (Option memory optionInfo) {
-        (uint160 optionId,) = getDecodedIdComponents(tokenId);
-        optionInfo = _option[optionId];
+        (uint160 optionKey,) = decodeTokenId(tokenId);
+        optionInfo = _option[optionKey];
     }
 
     /// @inheritdoc IOptionSettlementEngine
-    function claim(uint256 tokenId) external view returns (Claim memory claimInfo) {
+    function claim(uint256 tokenId) external view returns (OptionLotClaim memory claimInfo) {
         claimInfo = _claim[tokenId];
     }
 
-    function claimBucket(uint256 optionId, uint16 dayBucket)
-        external
-        view
-        returns (ClaimBucket memory claimBucketInfo)
-    {
-        (uint160 _optionId,) = getDecodedIdComponents(optionId);
-        claimBucketInfo = _claimBucketByOption[_optionId][dayBucket];
+    /// @inheritdoc IOptionSettlementEngine
+    function underlying(uint256 tokenId) external view returns (Underlying memory underlyingPositions) {
+        (uint160 optionKey, uint96 claimNum) = decodeTokenId(tokenId);
+
+        if (!isOptionInitialized(optionKey)) {
+            revert TokenNotFound(tokenId);
+        }
+
+        Option storage optionRecord = _option[optionKey];
+
+        // token ID is an option
+        if (claimNum == 0) {
+            bool expired = (optionRecord.expiryTimestamp > block.timestamp);
```

Comment:
this is a bug right here fixed in the other branch but since this method is moved, it wasn't ported. this needs to be geq, not lt